tests/tests/test_demo_artifact.py

```diff
@@ -79,31 +89,57 @@ def run_demo_script_up():
     def test_demo_artifact(self, run_demo_script):
         """Tests that the demo script does indeed upload the demo Artifact to the server."""
 
+        # Give the tests a clean docker-compose environment
         stop_docker_compose()
 
         logging.info("--------------------------------------------------")
         logging.info("Running test_demo_artifact_upload")
         logging.info("--------------------------------------------------")
-        self.demo_artifact_upload(run_demo_script)
-        stop_docker_compose()
+        demo_script = run_demo_script()
+        try:
+            self.demo_artifact_upload(demo_script)
+        except Exception as e:
+            demo_script.kill()
+            raise e
         self.auth.reset_auth_token()
+        stop_docker_compose()
 
         logging.info("--------------------------------------------------")
         logging.info("Running test_demo_artifact_installation")
         logging.info("--------------------------------------------------")
-        self.demo_artifact_installation(run_demo_script)
-        stop_docker_compose()
+        demo_script = run_demo_script()
+        try:
+            self.demo_artifact_installation(demo_script)
+        except Exception as e:
+            demo_script.kill()
+            raise e
         self.auth.reset_auth_token()
+        stop_docker_compose()
 
         logging.info("--------------------------------------------------")
         logging.info("Running test_demo_up_down_up")
         logging.info("--------------------------------------------------")
-        self.demo_up_down_up(run_demo_script)
-        stop_docker_compose()
+        demo_script = run_demo_script()
+        try:
+            self.demo_artifact_upload(demo_script)
+        except Exception as e:
+            demo_script.kill()
+            raise e
+
+        try:
+            demo_script = run_demo_script()
+            self.demo_artifact_upload(demo_script)
+        except DemoScriptException as e:
+            if e.err_code != 0:
+                raise e
+            else:
+                pass
+        except Exception as e:
+            raise e
```

Comment:
This last exception is not necessary, by "default" it will raise everything that you don't except